core/driver/sources/src/main/scala/org/virtuslab/ideprobe/dependencies/IntelliJResolver.scala

```diff
@@ -37,9 +43,54 @@ case class IntelliJPatternResolver(pattern: String) extends IntelliJResolver {
       "version" -> version.releaseOrBuild,
       "release" -> version.release.getOrElse("snapshot-release")
     )
-    val replaced = replacements.foldLeft(pattern) { case (path, (pattern, replacement)) =>
+    val replacedBeforeResolvingGlobs = replacements.foldLeft(pattern) { case (path, (pattern, replacement)) =>
       path.replace(s"[$pattern]", replacement)
     }
+    val replaced =
+      if (replacedBeforeResolvingGlobs.startsWith("file:"))
+        resolveGlobsInPattern(replacedBeforeResolvingGlobs)
+      else
+        replacedBeforeResolvingGlobs
+
     Dependency(replaced)
   }
+
+  private def resolveGlobsInPattern(pathPattern: String): String =
+    replaceGlobsWithExistingDirectories(List(pathPattern), pathPattern).head
+
+  // Solution below assumes that each * character is used to mark one part of the path (one atomic directory),
+  // for example: "file:///home/.cache/ides/com.jetbrains.intellij.idea/ideaIC/[revision]/*/ideaIC-[revision]/".
+  // Wildcard character should NOT be used as the last element of the pattern - to avoid ambiguous results.
```

Comment:
Hmmm technically, ambiguous results can also happen when wildcard is somewhere inside the path as well, right?

Maybe sth like:
```suggestion
  // Wildcard character should NOT be used as the last element of the pattern - to decrease the probability of getting ambiguous results.
```

Reply:
> ambiguous results can also happen when wildcard is somewhere inside the path as well, right?

No, if the `*` wildcard is used in replacement of an atomic directory from the path (but not the last part) - then 
```
private def resolveGlobsInPattern(pathPattern: String): String =
    replaceGlobsWithExistingDirectories(List(pathPattern), pathPattern).head
```
- will return only one element if pattern is valid (so the result will not be ambiguous)
or
- will throw `NoSuchElementException` if pattern is not valid

Reply:
Huh okay... so let's break down all cases:

1. zero glob matches: exception regardless of where the glob is located?
2. exactly one glob match: OK
3. more than one glob match:
  a. if `*` is used for the last part of the glob &mdash; multiple results returned?
  b. if `*` is used for other part of the glob &mdash; exception thrown?

Reply:
Answered in PMs but short answers below:

> 1. zero glob matches: exception regardless of where the glob is located?

`NoSuchElementException` will be thrown (as pattern is not valid - such file does not exist)

> 2. exactly one glob match: OK

yes

> 3. more than one glob match:
a. if * is used for the last part of the glob — multiple results returned?

**no** - `NoSuchElementException` will be thrown (as pattern is not valid - such file does not exist) - using `*` as the last element of the path is not possible, because then the logic would not be able to choose the proper path.
For example - if we would have to path endings like below:
```
.../df364b7662aeb20ca2eafead730380017c395661/ideaIC-223.4884.69-EAP-SNAPSHOT 
.../df364b7662aeb20ca2eafead730380017c395661/ideaIC-223.4884.69-EAP-SNAPSHOT.zip
.../df364b7662aeb20ca2eafead730380017c395661/ideaIC-223.4884.69-EAP-SNAPSHOT.jar
.../df364b7662aeb20ca2eafead730380017c395661/ideaIC-223.4884.69-EAP-SNAPSHOT.pom
```
where first one is directory and second one - a `.zip` file - and the pattern ends with `*` - how can we know, which one should be used?

> 3. more than one glob match:
b. if * is used for other part of the glob — exception thrown?

**no** - `resolveGlobsInPattern` will choose the first one by calling `.head`